docs/DeveloperGuide.adoc

```diff
@@ -453,6 +453,97 @@ does not have to be aware of new `Predicates` +
+Unify have a built-in background music player. Users have the choice of playing some music to relax their mind and reduce their stress.
+
+The `music` command is extended with an additional command behind it. For example, `music play`, has play as the extended command, this will `play` command at the back means that the system will play a music track. While `music pause` will pause the current music track.
+
+All these different actions are executed in the same `MusicCommand` class. This keep the commands easier to remember for the user as well, as anything related to `music` will start with `music` as a command.
+
+As shown below, additional extended commands can be added by having an additional case in the switch statement, as the extended commands are handled in a switch statement.
+
+[source,java]
+----
+public CommandResult execute() {
+        boolean genreExist = Arrays.asList(genreList).contains(genre);
+        switch (command) {
+        case "play":
+            if (mediaPlayer != null && mediaPlayer.getStatus() == MediaPlayer.Status.PAUSED) {
+                mediaPlayer.play();
+                return new CommandResult(MESSAGE_SUCCESS);
+            }
+            if (genreExist) {
+                String musicFile = "audio/music/" + genre + trackNumber + ".mp3";
+                MESSAGE_SUCCESS = genre.toUpperCase() + " Music Playing";
+                if (trackNumber < 2) {
+                    trackNumber++;
+                } else {
+                    trackNumber = 1;
+                }
+                if (mediaPlayer != null && mediaPlayer.getStatus() == MediaPlayer.Status.PLAYING) {
+                    mediaPlayer.stop();
+                }
+                Media sound = new Media(new File(musicFile).toURI().toString());
+                mediaPlayer = new MediaPlayer(sound);
+                mediaPlayer.setCycleCount(MediaPlayer.INDEFINITE);
+                mediaPlayer.setVolume(2.0);
+                mediaPlayer.play();
+                return new CommandResult(MESSAGE_SUCCESS);
+            } else {
+                return new CommandResult(MESSAGE_USAGE);
+            }
+        case "stop":
+            if (mediaPlayer != null && mediaPlayer.getStatus() == MediaPlayer.Status.PLAYING) {
+                mediaPlayer.stop();
+            }
+            return new CommandResult(MESSAGE_MUSICSTOP);
+        case "pause":
+            if (mediaPlayer != null && mediaPlayer.getStatus() == MediaPlayer.Status.PLAYING) {
+                mediaPlayer.pause();
+            }
+            MESSAGE_MUSICPAUSE = genre.toUpperCase() + " Music Paused";
+            return new CommandResult(MESSAGE_MUSICPAUSE);
+        default:
+            return new CommandResult(MESSAGE_USAGE);
+        }
+    }
+
+----
+[NOTE]
+The `trackNumber` has to be equal or less than the number of each genre tracks you have in the `audio/music` folder. For example, if `trackNumber` is 2, it is required at least 2 mp3 files of each genre of music.
```

Comment:
The `trackNumber` has to be **less than or equal** to the number of genre tracks you have in the `audio/music` folder. For example, if `trackNumber` is 2, you will require at least 2 .mp3 files of each genre of music.